Add support for per API configurable options

maririos opened the pull request:
- Creates an `{operation}Options` object per `operation`
  - RecognizeCustomFormOptions
  - RecognizeContentOptions
  - RecognizeReceiptOptions 
  - TrainingOptions
- Each `option` is located next to the client namespace
- Moved `ContentType` to main namespace as it is an input parameter and now encapsulated by the other `Option` objects

API view: https://apiview.dev/Assemblies/Review/cf2c57b3d4c44ef08c2e8284f7441a2d?diffRevisionId=0ee80a9f9a204a379ffca0e21f251fd0

Fixes: https://github.com/Azure/azure-sdk-for-net/issues/14097

maririos committed:
creating options

maririos committed:
fix comment

maririos committed:
add missing namespace change

kristapratico reviewed (approved):
Looks good. Does Changelog need update?

samvaity commented on sdk/formrecognizer/Azure.AI.FormRecognizer/src/RecognizeCustomFormsOptions.cs:
```diff
@@ -1,19 +1,18 @@
+    /// to configure the behavior of the request.
     /// </summary>
-    public class RecognizeOptions
+    public class RecognizeCustomFormsOptions
```
@maririos Was curious if we should match this with Content options naming and keep it as `RecognizeCustomFormOptions` or was the idea to keep it consistent with method naming?

maririos replied:
yes. consistency with method naming as we use `operation` in both method and options name

maririos commented:
>Does Changelog need update?

Yes! thank you for the reminder. Will add

kinelski commented on sdk/formrecognizer/Azure.AI.FormRecognizer/src/RecognizeContentOptions.cs:
```diff
@@ -0,0 +1,25 @@
+namespace Azure.AI.FormRecognizer
+{
+    /// <summary>
+    /// The set of options that can be specified when calling a recognition content method
```
nit: I think `a content recognition` method makes more sense, but take my English suggestions with a grain of salt.

kinelski replied:
Or just saying `calling a recognize content method` like you did in the Receipts Options sounds good as well.

kinelski commented on sdk/formrecognizer/Azure.AI.FormRecognizer/src/RecognizeCustomFormsOptions.cs:
```diff
@@ -1,19 +1,18 @@
-    /// The set of options that can be specified when calling a recognition method in
-    /// a <see cref="FormRecognizerClient" /> instance to configure the behavior of the
-    /// request.
+    /// The set of options that can be specified when calling a recognize custom form method
```
nit: `recognize custom form method` => `recognize custom forms method`

Too picky, I know.

kinelski commented on sdk/formrecognizer/Azure.AI.FormRecognizer/src/FieldData.cs:
```diff
@@ -25,7 +25,7 @@ internal FieldData(string text, int pageNumber, BoundingBox boundingBox, IReadOn
 
         /// <summary>
-        /// When <see cref="RecognizeOptions.IncludeFieldElements"/> is set to <c>true</c>, a list of references to
+        /// When `IncludeFieldElements` is set to <c>true</c>, a list of references to
```
Took me a while to understand the need of this change.

I would prefer using single quote marks `'IncludeFieldElements'`, but I don't feel strongly about this one. Just a suggestion, so feel free to skip it.

maririos committed:
add changelog

kinelski commented on sdk/formrecognizer/Azure.AI.FormRecognizer/CHANGELOG.md:
```diff
@@ -6,6 +6,8 @@
 - The library now targets the service's v2.0 API, instead of the v2.0-preview.1 API.
 - Updated version number from `1.0.0-preview.5` to `3.0.0-preview.5`.
+- Added models `RecognizeCustomFormsOptions`, `RecognizeReceiptsOptions`, and `RecognizeContentOptions` instead of a generic `RecognizeOptions` to support passing configurable options to recognize API's.
+- Added model`TrainingOptions` to support passing configurable options to training API's. This type now includes `TrainingFileFilter`.
```
nit: missing blank space between `model` and `TrainingOptions`.

kinelski replied:
nit: regarding the two additions to the changelog, I believe [we should drop the apostrophe when pluralizing acronyms](https://www.wikihow.com/Pluralize-Abbreviations,-Acronyms,-and-Initialisms). So: `APIs` instead of `API's`. I think a lot of people use the apostrophe anyway, so I wouldn't call this an issue.

kinelski reviewed (approved):
I know you just merged it, but here are my nitpicking comments for a next PR, maybe.

maririos commented:
>I know you just merged it, but here are my nitpicking comments for a next PR, maybe.

Thank you!!! I addressed all your comments in PR https://github.com/Azure/azure-sdk-for-net/pull/14106
